stores/ApplicationStore.js

```diff
@@ -5,11 +5,15 @@ class ApplicationStore extends BaseStore {
     constructor(dispatcher) {
         super(dispatcher);
         this.pageTitle = '';
+        this.frozen = false;
     }
     handlePageTitle(payload) {
         this.dispatcher.waitFor(RouteStore, () => {
-            this.pageTitle = payload.pageTitle;
-            this.emitChange();
+            if (!this.frozen || payload.allowUnfreeze) {
```

Comment:
Can anyone else also take a look at this? seems a bit tricky, but if it does the job I think it is good.